hoomd/md/minimize/fire.py

```diff
@@ -0,0 +1,191 @@
+    stays the same), the velocity of the particles is set to 0, :math:`\\alpha
+    \\rightarrow \\ \\alpha_{start}` and :math:`\\delta t \\rightarrow \\delta t
+    \\cdot fdec_{\\alpha}`. Convergence is determined by both the force per
+    particle and the change in energy per particle dropping below *angmom_tol*
+    and *energy_tol*, respectively or
+
+    .. math::
+
+        \\frac{\\sum |F|}{N*\\sqrt{N_{dof}}} < angmom_{tol} \\;\\; and \\;\\
+        \\Delta \\frac{\\sum|E|}{N} < energy_{tol}
+
+    where N is the number of particles the minimization is acting over (i.e.
+    the group size) Either of the two criterion can be effectively turned off
+    by setting the tolerance to a large number.
+
+    If the minimization is acted over a subset of all the particles in the
+    system, the "other" particles will be kept frozen but will still interact
+    with the particles being moved.
+
+    Examples::
+
+        fire = md.minimize.FIRE(dt=0.05)
+        fire.force_tol = 1e-2
+        fire.energy_tol = 1e-7
+        fire.methods.append(methods.NVE(filter.All()))
+        sim.operations.integrator = fire
+        while not(fire.has_converged()):
+           sim.run(100)
+
+    Examples::
+
+        fire = md.minimize.FIRE(dt=0.05)
+        fire.methods.append(methods.NPH(filter.All(), P=0.0, gamma=.5))
+        sim.operations.integrator = fire
+        while not(fire.has_converged()):
+           sim.run(100)
+
+    Note:
+        The algorithm requires a base integrator to update the particle
+        position and velocities. Usually this will be either NVE (to minimize
+        energy) or NPH (to minimize energy and relax the box). The quantity
+        minimized is in any case the energy (not the enthalpy or any other
+        quantity).
+
+    Note:
+        As a default setting, the algorithm will start with a :math:`\\delta t
+        = \\frac{1}{10} \\delta t_{max}` and attempts at least 10 search steps.
+        In practice, it was found that this prevents the simulation from making
+        too aggressive a first step, but also from quitting before having found
+        a good search direction. The minimum number of attempts can be set by
+        the user.
+
+    """
+    _cpp_class_name = "FIREEnergyMinimizer"
+
+    def __init__(self,
+                 dt,
+                 aniso='auto',
+                 forces=None,
+                 constraints=None,
+                 methods=None,
+                 rigid=None):
+
+        super().__init__(forces, constraints, methods, rigid)
+
+        self._param_dict.update(
+            ParameterDict(
+                dt=float(dt),
+                aniso=OnlyFrom(['true', 'false', 'auto'],
+                               preprocess=_preprocess_aniso),
+                min_steps_adapt=int,  # TODO force positive integers
+                finc_dt=float,
+                fdec_dt=float,
+                alpha_start=float,  # TODO prevent user from resetting value
```

Comment:
What I really want to avoid here is someone resetting this value in the middle of a simulation run. Maybe it would be better to make it a parameter of the constructor and then override `_setattr_param`.

Reply:
After thinking more, I've reconsidered. This value should be allowed to change.